app/src/main/java/org/wikipedia/login/LoginClient.kt

```diff
@@ -1,24 +1,21 @@
 package org.wikipedia.login
 
 import android.widget.Toast
-import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
-import io.reactivex.rxjava3.core.Observable
-import io.reactivex.rxjava3.disposables.CompositeDisposable
-import io.reactivex.rxjava3.schedulers.Schedulers
-import kotlinx.serialization.SerialName
-import kotlinx.serialization.Serializable
+import androidx.lifecycle.LifecycleCoroutineScope
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
 import org.wikipedia.R
 import org.wikipedia.WikipediaApp
 import org.wikipedia.dataclient.Service
 import org.wikipedia.dataclient.ServiceFactory
 import org.wikipedia.dataclient.WikiSite
-import org.wikipedia.dataclient.mwapi.MwResponse
 import org.wikipedia.util.log.L
 import java.io.IOException
 
 class LoginClient {
 
-    private val disposables = CompositeDisposable()
+    private var clientJob: Job? = null
```

Comment:
@dbrant Do we still need this `Job` for canceling the current request? It will be called when the `LoginActivity` on `onStop`. I wonder if we can let the activity lifecycle decide whether to cancel the `coroutine` process or do it explicitly.

Reply:
Thanks! Done.